core/storage/btree.rs

```diff
@@ -78,6 +83,10 @@ impl BTreeCursor {
         }
     }
 
+    fn cell_count(&self, page_idx: usize) -> Result<CursorResult<usize>> {
```

Comment:
IMO it's a bit surprising that a function called `cell_count()` would implicitly read a page (possibly doing I/O) and return the count. Can we do the read separately instead? This function is also only used in one place as far as I can tell.